packages/app/src/pages/Specs/Runner.vue

```diff
@@ -48,18 +48,53 @@ subscription SpecPageContainer_specsChange {
 }
 `
 
-useSubscription({ query: SpecPageContainer_SpecsChangeDocument })
+gql`
+subscription Runner_ConfigChange {
+  configChange {
+    id
+    ...SpecRunner_Config
+  }
+}
+`
 
 const isRunMode = window.__CYPRESS_MODE__ === 'run'
 
+// subscriptions are used to trigger live updates without
+// reloading the page.
+// this is only useful in open mode - in run mode, we don't
+// use GraphQL, so we pause the
+// subscriptions so they never execute.
+const shouldPauseSubscriptions = isRunMode && window.top === window
+
+useSubscription({
+  query: SpecPageContainer_SpecsChangeDocument,
+  pause: shouldPauseSubscriptions,
+})
+
 // in run mode, we are not using GraphQL or urql
 // for performance - run mode does not need the
 // same level of runner interactivity as open mode.
 // by setting `pause: true` in run mode, urql will not trigger any
 // requests, which is what we want.
 const query = useQuery({
   query: SpecPageContainerDocument,
-  pause: isRunMode && window.top === window,
+  pause: shouldPauseSubscriptions,
+})
+
+let initialLoad = true
+
+useSubscription({ query: Runner_ConfigChangeDocument }, () => {
+  // if the `config` changed, we want to reload the entire
+  // page and re-execute the current test with the latest config
+  // values
+  // subscriptions trigger on the initial page load,
+  // so we do not want to trigger `window.location.reload` on the
+  // first load, or we get stuck in an infinite loop.
```

Comment:
What if we send the config values in the subscription as a new field with a `JSON` payload, and then set them on `window.__CYPRESS_CONFIG__`?

Reply:
We could, but those are base64 encoded right now, we'd need to re-encode it as base64, or receive it as base64.

I'm not super happy we are sticking the config on `window`, but I guess it's probably the most reliable way to do this right now.

I'll give this a try.

Reply:
Your idea is quite a bit better in terms of reliability. I found something troubling; the config we source in `data-context` was missing some dynamic values like the `proxyUrl`. For now I patched it in, but we need to continue to aggressively triage the config problem we've got and find a single way to manage and source it. 

I'd say the next step would be a solid refactor of `project-base`, `open-project` and `server-{base,e2e,ct}` and the surrounding area, but this is quite high risk, so not sure we should do that right now. I might start a tech brief where we can gather our thoughts about this.